cassandra/common/src/main/java/org/opennms/newts/cassandra/CassandraSessionImpl.java

```diff
@@ -15,115 +15,144 @@
+    public CassandraSessionImpl(@Named("cassandra.localdatacenter") String localDatacenter,
+            @Named("cassandra.keyspace") String keyspace, @Named("cassandra.hostname") String hostname,
             @Named("cassandra.port") int port, @Named("cassandra.compression") String compression,
             @Named("cassandra.username") String username, @Named("cassandra.password") String password,
             @Named("cassandra.ssl") boolean ssl,
+            @Deprecated
             @Named("cassandra.pool.core-connections-per-host") Integer coreConnectionsPerHost,
             @Named("cassandra.pool.max-connections-per-host") Integer maxConnectionsPerHost,
-            @Named("cassandra.pool.max-requests-per-connection") Integer maxRequestsPerConnection) {
+            @Named("cassandra.pool.max-requests-per-connection") Integer maxRequestsPerConnection,
+            @Named("cassandra.driver-settings-file") String driverSettingsFile) {
+
+        if (driverSettingsFile != null) {
+            File settingsFile = new File(driverSettingsFile);
+            LOG.info("Setting up session with settings file: {}", settingsFile);
+            m_session = CqlSession.builder()
+                    .withConfigLoader(DefaultDriverConfigLoader.fromFile(settingsFile))
+                    .build();
+            return;
+        }
 
+        checkNotNull(localDatacenter, "localDatacenter argument");
         checkNotNull(keyspace, "keyspace argument");
         checkNotNull(hostname, "hostname argument");
         checkArgument(port > 0 && port < 65535, "not a valid port number: %d", port);
         checkNotNull(compression, "compression argument");
 
-        LOG.info("Setting up session with {}:{} using compression {}", hostname, port, compression.toUpperCase());
+        LOG.info("Setting up session with to {}:{} using compression {} and local datacenter: {}", hostname, port,
+                compression.toUpperCase(), localDatacenter);
+
+        ProgrammaticDriverConfigLoaderBuilder configBuilder = DriverConfigLoader.programmaticBuilder()
+                .startProfile("default");
+        configBuilder.withString(DefaultDriverOption.SESSION_KEYSPACE, keyspace);
+        configBuilder.withStringList(DefaultDriverOption.CONTACT_POINTS, toContactPoints(hostname, port));
 
-        final PoolingOptions poolingOptions = new PoolingOptions();
         if (coreConnectionsPerHost != null) {
-            LOG.debug("Using {} core connections per host.", coreConnectionsPerHost);
-            poolingOptions.setCoreConnectionsPerHost(HostDistance.LOCAL, coreConnectionsPerHost)
-                    .setCoreConnectionsPerHost(HostDistance.REMOTE, coreConnectionsPerHost);
+            LOG.warn("Ignoring core connections per host value of: {}. " +
+                    "Use 'max-connections-per-host' only instead.", coreConnectionsPerHost);
+
         }
```

Comment:
Then why keep it at all? It's 3.0.0, it's reasonable to expect users to switch to a new signature.

Reply:
Good point - I'll revisit once I work through the configuration on the OpenNMS side